src/basics.ml

```diff
@@ -34,6 +34,18 @@ let to_tvar : term -> tvar = fun t ->
     “marshaled” (e.g., by the {!module:Sign} module), as this would break the
     freshness invariant of new variables. *)
 
+(** [stamp_tvar s v] creates a fresh variable whose name is the name of [v]
```

Comment:
It is related to #233:
with
```
rule f (λ x, g _) → g o
compute f (λ x, f (λ t, g x))
```
I was afraid that if I don't distinguish manually variables between substitution, the `x` and the `t` would be seen as the the same.  Indeed, to match ` f (λ t, g x)`, the same tree is used, and thus possibly the same variable.

Reply:
I see, but I'm not sure this is useful. Probably not since the two binders will be destructed with `Bindlib.unbind` (I guess), and this creates (distinct) fresh variables. So I think you can just remove all that.

If I'm wrong, then for sure you don't need the mechanism you implemented to do what you want. You (at most) need something like the following to get a fresh variables that is distinct from `x` (or any other variable) but has the same name.
```ocaml
let y = Bindlib.new_var mkfree (Bindlib.name_of x) in
...
```
No need to manage a counter yourself, Bindlib will do the minimal renaming require to avoid ambiguity when printing.

Reply:
Right I have to rethink that part a bit

Reply:
I can't use `unbind` because I need some consistence between the allowed bound variables in a pattern variable and the substituted variables.  The former are accessible at compile time and the latter during evaluation.  So I create fresh variables at compile time, and I reuse them during evaluation.

Reply:
OK, I see. Then at least use `new_var` to create a fresh variable (as explained above) and replace the "stamping" thing.

Reply:
I removed that stamping process, which revealed some bugs, which have been fixed, which raised issue #255 in file `ho_bug1.dk`, which I thus fixed using as a model your PR.  It should be good now.